src/Sulu/Component/Content/SmartContent/QueryBuilder.php

```diff
@@ -359,7 +359,12 @@ private function buildTypesWhere($types, string $languageCode)
     {
         $sql2Where = [];
         foreach ($types as $type) {
-            $sql2Where[] = 'page.[i18n:' . $languageCode . '-template] = ' . $type;
+            if ('snippet' === static::$structureType) {
```

Comment:
Yes you are right, but this would require bigger changes and most likely BC breaks. I already discussed it with @wachterjohannes and we didn't had a better idea.

Reply:
Why can't we just override the `buildTypesWhere` function in the `SnippetQueryBuilder`?

Reply:
Because the `buildTypes..` methods are private, but we could make them protected and override it.